src/gui/plugins/axes/AxesConfig.cc

```diff
@@ -0,0 +1,285 @@
+    // rendering calls here
+    UpdateOriginArrows();
+  }
+
+  // Standard event processing
+  return QObject::eventFilter(_obj, _event);
+}
+
+void AxesConfig::UpdateActiveAxes()
+{
+  for(auto vis : this->dataPtr->activeAxesMap)
+  {
+    auto visAxes = std::dynamic_pointer_cast<rendering::AxisVisual>(
+          this->dataPtr->scene->VisualByName(vis.first + "Axes"));
+
+    auto visEntity = std::dynamic_pointer_cast<rendering::Visual>(
+          this->dataPtr->scene->VisualByName(vis.first));
+
+    if (visEntity && visAxes){
+      math::Pose3d pose_entity = visEntity->LocalPose();
+      visAxes->SetLocalPose(vis.second + pose_entity);
+    }
+  }
+}
+
+/////////////////////////////////////////////////
+void AxesConfig::UpdateOriginArrows()
+{
+  // Load axes if they don't already exist
+  this->LoadAxesbyName(this->dataPtr->name_axes);
+
+  // If axes were not loaded successfully, don't update
+  if (!this->dataPtr->axes)
+    return;
+
+  // Save the axesVisual in the structure if it doesn't exist or update the pose
+  auto it = this->dataPtr->activeAxesMap.find(this->dataPtr->name_axes);
+  if (it == this->dataPtr->activeAxesMap.end())
+  {
+    this->dataPtr->activeAxesMap.insert(
+      std::pair<std::string, math::Pose3d>(this->dataPtr->name_axes, this->dataPtr->pose));
+  } else {
+    it->second = this->dataPtr->pose;
+  }
+
+  // Updating the poses
+  this->UpdateActiveAxes();
+
+  // update visibility
+  this->dataPtr->axes->SetVisible(this->dataPtr->visible);
+  // Update type: arrow or line
+  if (this->dataPtr->visible)
+  {
+    this->dataPtr->axes->ShowAxisHead(this->dataPtr->isArrow);
+  }
+  // update scale
+  this->dataPtr->axes->SetLocalScale(1, 1, this->dataPtr->length * 2);
+}
+
+const QStringList AxesConfig::comboList()
+{
+    return m_comboList;
+}
+
+void AxesConfig::setComboList(const QStringList &comboList)
+{
+   if (m_comboList != comboList)
+   {
+       m_comboList = comboList;
+       emit comboListChanged();
+   }
+}
+
+void AxesConfig::LoadAxesbyName(const std::string & name)
+{
+  if (this->dataPtr->name_axes.compare(name) == 0 && this->dataPtr->axes)
+    return;
+
+  this->dataPtr->name_axes = name;
+  if (this->dataPtr->name_axes.compare(worldOriginAxes) != 0)
+  {
+    this->dataPtr->axes = std::dynamic_pointer_cast<rendering::AxisVisual>(
+        this->dataPtr->scene->VisualByName(this->dataPtr->name_axes + "Axes"));
+  } else {
+    this->dataPtr->axes = std::dynamic_pointer_cast<rendering::AxisVisual>(
+        this->dataPtr->scene->VisualByName(worldOriginAxes));
+  }
+
+  if (!this->dataPtr->axes)
+  {
+    rendering::VisualPtr root = this->dataPtr->scene->RootVisual();
+
+    auto axes = this->dataPtr->scene->CreateAxisVisual(this->dataPtr->name_axes + "Axes");
+    root->AddChild(axes);
+  }
+}
+
+void AxesConfig::onCurrentIndexChanged(int _index)
+{
+  std::string name_entity = m_comboList[_index].toStdString();
+  LoadAxesbyName(name_entity);
+}
+
+void AxesConfig::EntitiesInScene()
```

Comment:
@chapulina, Is there other way to get the entitie names from a plugin?

Reply:
If you change the plugin to inherit from `gazebo::GuiSystem`, then you could use an `Each` call to get for example the names of all visuals, links, collisions, models, etc. The downside is that this plugin would become Gazebo-specific so we can't port it to `ign-gui` later.